File: 937 reorder log in log files/Reorder.java
public class Reorder {
    public String[] reorderLogFiles(String[] logs) {
        Arrays.sort(logs, (i, j) -> {
            String[] info1 = i.split(" ");
            String[] info2 = j.split(" ");
            String sub1 = i.substring(info1[0].length() + 1);
            String sub2 = j.substring(info2[0].length() + 1);
            if (isLetter(info1[1].charAt(0)) && isLetter(info2[1].charAt(0))) {
                if (sub1.equals(sub2)) {
                    return i.compareTo(j);
                }
                return sub1.compareTo(sub2);
            } else if (isDigit(info1[1].charAt(0)) && isDigit(info2[1].charAt(0))) {
                return 0;
            } else {
                return isDigit(info1[1].charAt(0)) ? 1 : -1;
            }
        });
        return logs;
    }
    
    private boolean isLetter(char c) {
        return c <= 'z' && c >= 'a';
    }
    
    private boolean isDigit(char c) {
        return c <= '9' && c>= '0';
    }
}

File: 937 reorder log in log files/Reorder.py
class Solution(object):
    def reorderLogFiles(self, logs):
        """
        :type logs: List[str]
        :rtype: List[str]
        """
        digits = []
        letters = []
        
        for log in logs:
            if log.split()[1].isdigit():
                digits.append(log)
            else:
                letters.append(log)
        

        # Has to sort by identifier first, because we need to make sure
        # we sort by log when logs don't tie
        letters.sort(key = lambda x : x.split()[0])  # sort by identifier when log ties
        letters.sort(key = lambda x : x.split()[1:]) # sort by log
        
        return letters + digits